VotingApplication/VotingApplication.Data/Model/Event.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+
+namespace VotingApplication.Data.Model
+{
+    public enum EventType
+    {
+        Error,
+        GoToPage,
+        CreatePoll,
+        UpdateResults,
+        SetExpiry,
+        SetPollType,
+        SetQuestion,
+        SetInviteOnly,
+        SetNamedVoting,
+        SetOptionAdding,
+        AddOption,
+        UpdateOption,
+        DeleteOption,
+        AddVote,
+        DeleteVote,
+        AddBallot,
+        DeleteBallot,
+        Login,
+        Logout,
+        Register
+    }
+
+    public class Event
+    {
+        public Event(EventType eventType, Guid pollId)
+        {
+            Timestamp = DateTime.Now;
+            EventType = eventType.ToString();
+            PollId = pollId;
+        }
+
+        public long Id { get; set; }
+
+        [Required]
+        public DateTime Timestamp { get; set; }
+
+        [Required]
+        public string EventType { get; set; }
```

Comment:
I can see the logic, but strings are horrible, really. I don't think it's a particularly good argument that it would be more awkward with a join table, as it's pretty trivial to implement. I would argue that the integrity you get with storing the enum value more than outweighs the convenience of not having to have a join table.